ntp-udp/src/lib.rs

```diff
@@ -221,14 +282,93 @@ fn recv(socket: &std::net::UdpSocket, buf: &mut [u8]) -> io::Result<(usize, Opti
     Ok((bytes_read, None))
 }
 
+fn fetch_send_timestamp_help(
+    socket: &std::net::UdpSocket,
+    expected_counter: u32,
+) -> io::Result<Option<NtpTimestamp>> {
+    // we get back two control messages: one with the timestamp (just like a receive timestamp),
+    // and one error message with no error reason. The payload for this second message is kind of
+    // undocumented.
+    //
+    // section 2.1.1 of https://www.kernel.org/doc/Documentation/networking/timestamping.txt says that
+    // a `sock_extended_err` is returned, but in practice we also see a socket address. The linux
+    // kernel also has this https://github.com/torvalds/linux/blob/master/tools/testing/selftests/net/so_txtime.c#L153=
+    const CONTROL_SIZE: usize = control_message_space::<[libc::timespec; 3]>()
+        + control_message_space::<(libc::sock_extended_err, libc::sockaddr_storage)>();
```

Comment:
This seems to me to be somewhat weird, and doesn't entirely match with what the selftests do either. Are we sure this is correct? I would expect more something along the line of:
`control_message_space::<libc::timespec>` for the timestamp
`+control_message_space::<libc::sock_extended_err>` for the error part
`+control_message_space::<libc::sockaddr_storage>` for the sockaddr

Reply:
I've adjusted this to 
```
    const CONTROL_SIZE: usize = control_message_space::<[libc::timespec; 3]>()
        + control_message_space::<(libc::sock_extended_err, libc::sockaddr)>();
```

the value of that is 112, which is exactly how many bytes are used (based on mhdr.msg_controllen after the receive_message. I checked this with a bigger buffer: 112 is what is actually used).

given these sizes in practice

```
[ntp-udp/src/lib.rs:301] control_message_space::<[libc::timespec; 3]>() = 64
[ntp-udp/src/lib.rs:301] control_message_space::<[libc::timespec; 1]>() = 32
[ntp-udp/src/lib.rs:301] control_message_space::<libc::sock_extended_err>() = 32
[ntp-udp/src/lib.rs:301] control_message_space::<libc::sockaddr>() = 32
```

I think my code is now correct and a tight bound.